scheduler/src/scheduler.cpp

```diff
@@ -60,372 +62,19 @@ Scheduler::Scheduler(vector<Task *> * tasks, double ** duration_matrix, double m
-//   double max=0;
-//   double dist;
-//   for(int i=0; i< numTasks; i++)
-//   {
-//     for(int j=0; j< numTasks; j++)
-//     {
-//       if(i != j)
-//       {
-//         dist = DistWrapper::dist(tasksToS->at(i)->getEndPos(),tasksToS->at(j)->getStartPos());
-//         duration_matrix[i][j] = dist;
-//         if(dist>max)
-//         {  
-//           max = dist; 
-//         }
-//       }
-//     }
-//   }
-//   return max;
-// }
-
 /**
-  This method is used during experiments. It reads the array of distances from the file. It is hard-coded for specific datasets G4S or AAF.
-  Not for public use.
-  @param none
-  @return maximal distance
+  Desctructor
 */
-double Scheduler::readDist() 
+Scheduler::~Scheduler()
 {
-  //string filename = "/home/lenka/phd/code/strands_ws/src/strands_executive/scheduler/data/distances_aaf.txt";
-  string filename = "/home/lenka/phd/code/strands_ws/src/strands_executive/scheduler/data/distances_g4s.txt";
-  ifstream results;
-  results.open (filename,std::ios_base::in);
-
-  double max=0;
-  double dist;
-  int numWay = 46; //or 17 for aaf 46 for g4s
-  double arr_dist[numWay][numWay];
-
-  double n;
-  int i=0,j=0;
-  while(results >> n)
-  {
-    arr_dist[i][j] = n;
-    j++;
-    if(n>max)
-        {  
-          max = n; 
-        }
-    if(j > numWay-1)
-    {
-      j = 0;
-      i++;
-    }
-  }
- results.close();
-
-  string point[] = {"Lobby", "ChargingPoint", "OfficeDoorInside", "OfficePoint", "OfficeDoorOutside", "SafePointLobby", "SafePointCorridor", "CrossRoads", "CorridorToilets", "GamePoint", "FireDoorWest", "FireDoorEast", "ChapelDoors", "FireExtinguisher", "PrinterRoom", "GameTable", "CorridorEast"};
-
-
-  for(int i=0; i< numTasks; i++)
-  {
-    for(int j=0; j< numTasks; j++)
-    {
-      if(i != j)
-      {
-        int x = -1,y = -1,sizex,sizey;
-        string EP = tasksToS->at(i)->getEndPos();
-        string SP = tasksToS->at(j)->getStartPos();
-
-        //aaf
-        /*for(int k = 0; k < numWay; k++)
-        {
-          if(EP == point[k])
-            x = k;
-          if(SP == point[k])
-            y = k;
-        }*/
- 
-        //g4s
-        sizex = EP.size();
-        sizey = SP.size();
-
-        if(sizex >=8)
-        {
-          if(EP == "ChargingPoint")
-            x = 0;
-          else if(EP.substr(0,8)=="WayPoint")
-          {
-            EP = EP.substr(8,sizex);  
-            x = stoi(EP);
-          }
-        }
-        if(sizey >= 8)
-        {
-          if(SP == "ChargingPoint")
-            y = 0;
-          else if(SP.substr(0,8)=="WayPoint")
-          {
-            SP = SP.substr(8,sizey);  
-            y = stoi(SP);
-          }
-        }
-
-        if((x != -1)&&(y != -1))
-          duration_matrix[i][j] = arr_dist[x][y];
-        
-      }
-       
-    }
-  }
-  
- 
-  
-  return max;
+  tasksToS = NULL;
+  numTasks = 0;
+  numPairs = 0;
+  duration_matrix = NULL; //but the matrix still exist in system, must be cleaned how call scheduler!
```

Comment:
It refers to the fact, that if I just set scheduler pointer to NULL, data still stays somewhere. But no worries, they are deleted in the scheduler_node :)